### PR TITLE
PR for Issue 21171: Add logic for getting claims from tokens

### PR DESCRIPTION
**Pull request by ayoho**
Adds initial logic for getting Caller Name and Caller Group claims from the access token or ID token, depending on where the claim is available.
    
For #21171

### CODE REVIEW

**Comment by ayoho** [comment]
#build

File: dev/io.openliberty.security.oidcclientcore.internal/src/io/openliberty/security/oidcclientcore/client/Client.java [hunk 1]
```diff
@@ -40,10 +46,10 @@ public ProviderAuthenticationResult continueFlow(HttpServletRequest request, Htt
         Flow flow = AbstractFlow.getInstance(oidcClientConfig);
         return flow.continueFlow(request, response);
     }
-    
-    public void validate(TokenResponse tokenResponse) throws TokenValidationException {
+
+    public JwtClaims validate(TokenResponse tokenResponse) throws TokenValidationException {
```
Comment (teddyjtorres) [inline dev/io.openliberty.security.oidcclientcore.internal/src/io/openliberty/security/oidcclientcore/client/Client.java:50]: Adding org.jose4j.jwt.JwtClaims caused a build failure at at io.openliberty.security.jakartasec.cdi.beans.OidcHttpAuthenticationMechanismTest.setUp(OidcHttpAuthenticationMechanismTest.java:90), where the Client class cannot be mocked due to a ClassNotFoundException:

Caused by: java.lang.ClassNotFoundException: org.jose4j.jwt.JwtClaims

The testpath in bnd.bnd for the io.openliberty.security.jakartasec.cdi will need to be updated to include jose4j.

File: dev/io.openliberty.security.jakartasec.3.0.internal/src/io/openliberty/security/jakartasec/identitystore/OidcIdentityStore.java [hunk 2]
```diff
@@ -48,4 +59,115 @@ public CredentialValidationResult validate(Credential credential) {
         return CredentialValidationResult.INVALID_RESULT;
     }
 
+    AccessToken createAccessTokenFromTokenResponse(TokenResponse tokenResponse) {
+        Map<String, String> tokenResponseRawMap = tokenResponse.asMap();
+        long expiresIn = 0L;
+        if (tokenResponseRawMap.containsKey(OpenIdConstant.EXPIRES_IN)) {
+            expiresIn = Long.parseLong(tokenResponseRawMap.get(OpenIdConstant.EXPIRES_IN));
+        }
+        // TODO - determine what value to use here
```
Comment (teddyjtorres) [inline dev/io.openliberty.security.jakartasec.3.0.internal/src/io/openliberty/security/jakartasec/identitystore/OidcIdentityStore.java:68]: Please use oidcClientConfig.getTokenMinValidity(). This is the value of the tokenMinValidity attribute, https://github.com/jakartaee/security/blob/master/api/src/main/java/jakarta/security/enterprise/authentication/mechanism/http/OpenIdAuthenticationMechanismDefinition.java#L397.

**Comment by teddyjtorres** [comment]
Rebased and delivered in https://github.com/OpenLiberty/open-liberty/pull/22577

**Comment by teddyjtorres** [comment]
Closing since these changes were already delivered in another PR.

**Pull request closed without merging**